cddl/ccf-merkle-tree-cose-signature.cddl

```diff
@@ -0,0 +1,28 @@
+ccf-cose-root-signature-tagged = #6.18(ccf-cose-root-signature)
+
+ccf-cose-root-signature = [
+  phdr : bstr .cbor protected_headers,  ; bstr-wrapped protected headers
+  uhdr : unprotected_headers,           ; no unprotected headers included
+  payload : nil,                        ; signed Merkle tree root hash, *detached* payload
+  signature : bstr                      ; COSE-signature
+]
+
+unprotected_headers = {
+  ; empty map instead of 'nil'. QCBOR library verifier doesn't support 'nil'.
+}
+
+protected_headers = {
+  &(alg: 1) => int,                     ; signing algoritm ID, as per RFC8152
+  &(kid: 4) => bstr .size 32,           ; signing key hash (string of HASH_SIZE(32) bytes)
```

Comment:
We may not want to hardcode that in case we need to change digest algos in the future, and have that be dependent on alg (it can/should still be 32 for the algs we support now).

Reply:
Ok, I'll remove the size check

Reply:
But wait, this's the key hash, we make it 32-bytes only, don't we?

Reply:
Talked offline, removed constraint as more flexible approach.